SSH and Passwd

SAtacker opened the pull request:
[![asciicast](https://asciinema.org/a/430542.svg)](https://asciinema.org/a/430542)

ArthurSonzogni commented:
Thanks!
Input elements seems to always appear as "Focused". I believe this was a bug I introduced recently and fixed in:
https://github.com/ArthurSonzogni/FTXUI/pull/188/commits/3657b3426aadb7867652c6c14b99fe92f67e0201
You probably want to refresh FTXUI version to the latest commit to avoid thi.

ArthurSonzogni commented on src/ui/panel/passwd/passwd.cpp:
```diff
@@ -0,0 +1,78 @@
+                        change_,
+                    }),
+                    [&] {
+                      if (view) {
```
I don't understand what the goal of "view" here. Could you please explain it to me? It seems this cause the layout to shift when you press enter.

SAtacker commented:
In case someone wants to verify what they have typed, a small text region
is useful.

On Mon, 16 Aug, 2021, 13:52 Arthur Sonzogni, ***@***.***>
wrote:

> ***@***.**** approved this pull request.
> ------------------------------
>
> In src/ui/panel/passwd/passwd.cpp
> <https://github.com/SAtacker/beagle-config/pull/36#discussion_r689330503>:
>
> > +    input_np = Input(&new_pass, "New Password", inp_opt);
> +    change_ = Button("Change", [&] {
> +      auto cmd = "passwd " + user;
> +      FILE* fp = popen(cmd.c_str(), "w");
> +      fprintf(fp, "%s\n", pass.data());
> +      fprintf(fp, "%s\n", new_pass.data());
> +      fprintf(fp, "%s\n", new_pass.data());
> +      pclose(fp);
> +    });
> +    page = Renderer(Container::Vertical({
> +                        input_p,
> +                        input_np,
> +                        change_,
> +                    }),
> +                    [&] {
> +                      if (view) {
>
> I don't understand what the goal of "view" here. Could you please explain
> it to me? It seems this cause the layout to shift when you press enter.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/SAtacker/beagle-config/pull/36#pullrequestreview-730445157>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AOXPF5P7W6C2BYKDRKZJH7TT5DDFPANCNFSM5CF77GAQ>
> .
>

SAtacker commented:
Thanks, I'll update it.

On Mon, 16 Aug, 2021, 13:47 Arthur Sonzogni, ***@***.***>
wrote:

> Thanks!
> Input elements seems to always appear as "Focused". I believe this was a
> bug I introduced recently and fixed in:
> ***@***.***
> <https://github.com/ArthurSonzogni/FTXUI/commit/3657b3426aadb7867652c6c14b99fe92f67e0201>
> You probably want to refresh FTXUI version to the latest commit to avoid
> thi.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/SAtacker/beagle-config/pull/36#issuecomment-899316890>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AOXPF5NSIRWJFSM3QWSGBQTT5DCRHANCNFSM5CF77GAQ>
> .
>
